### PR TITLE
Show welcome message and changelog URL after wrapper dist download

### PR DESCRIPTION
**Pull request by eriwen**
This change removes the wrapper logs for unzipping and setting
permissions, changing them to only log when something went wrong.

A welcome to Gradle message is displayed with a link to the releases
page for the new release.

Issue: #4623

### Before

```
Downloading https://services.gradle.org/distributions/gradle-4.6-bin.zip
........................................................................
Unzipping /Users/eric/.gradle/wrapper/dists/gradle-4.6-bin/6nscg4yape751pc62922nodvq/gradle-4.6-bin.zip to /Users/eric/.gradle/wrapper/dists/gradle-4.6-bin/6nscg4yape751pc62922nodvq
Set executable permissions for: /Users/eric/.gradle/wrapper/dists/gradle-4.6-bin/6nscg4yape751pc62922nodvq/gradle-4.6/bin/gradle
Starting a Gradle Daemon (subsequent builds will be faster)
Generating JAR file 'gradle-api-4.6.jar'
<other logs>
```

### After

```
Downloading https://services.gradle.org/distributions/gradle-4.7-bin.zip
........................................................................

Welcome to Gradle 4.7!
See what's new at https://gradle.org/releases/4.7

Starting a Gradle Daemon (subsequent builds will be faster)
Generating JAR file 'gradle-api-4.6.jar'
<other logs>
```

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java [hunk 1]
```diff
@@ -71,18 +89,28 @@ public File call() throws Exception {
...
                 markerFile.createNewFile();
 
+                printReleaseSummary();
+
                 return root;
             }
         });
     }
 
+    private void printReleaseSummary() {
```
Comment (bmuschko) [inline subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java]: :x: A null value for `gradleVersion` should be verified in `InstallTest.groovy`.

File: subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java [hunk 2]
```diff
@@ -71,18 +89,28 @@ public File call() throws Exception {
...
+        if (gradleVersion != null) { // appVersion could be null in custom distributions
+            logger.log("");
+            logger.log("Welcome to Gradle " + gradleVersion + "!");
```
Comment (bmuschko) [inline subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java]: :o: Maybe be easier to construct the message with a `{}` placeholder.

File: subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java [hunk 3]
```diff
@@ -71,18 +89,28 @@ public File call() throws Exception {
...
+            logger.log("See what's new at https://gradle.org/releases#" + gradleVersion);
```
Comment (bmuschko) [inline subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java]: :o: Would be awesome if we could make the URL clickable by default from the console (similar to what we do for failed test results).

Comment (bmuschko) [inline subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java]: :x: I believe that the message doesn't print the correct Gradle version. For example I built a local distribution, then used the distribution to create new Wrapper files for version 4.4. However, when running a task with the generated Wrapper files the version of my current distribution is used. This should be 4.4 and not 4.7-20180313213750+0000 right?

```
$ ./gradlew error
Downloading https://services.gradle.org/distributions/gradle-4.4-bin.zip
........................................................................

Welcome to Gradle 4.7-20180313213750+0000!
See what's new at https://gradle.org/releases#4.7-20180313213750+0000
```

File: subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java [hunk 4]
```diff
@@ -71,18 +89,28 @@ public File call() throws Exception {
...
 
+                printReleaseSummary();
+
                 return root;
             }
         });
     }
 
+    private void printReleaseSummary() {
+        if (gradleVersion != null) { // appVersion could be null in custom distributions
```
Comment (bmuschko) [inline subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java]: :x: We should only render the message for GA versions. For RCs we should use https://gradle.org/release-candidate/ instead or not render it at all. Snapshots do not have an equivalent and should not render the message.

File: subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java (hunk 3 above)
Comment (bmuschko) [inline subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java]: :o: I propose that we end the sentence with a dot.

File: subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperLoggingIntegrationTest.groovy [hunk 5]
```diff
@@ -17,18 +17,26 @@
 package org.gradle.integtests
 
 class WrapperLoggingIntegrationTest extends AbstractWrapperIntegrationSpec {
```
Comment (bmuschko) [inline subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperLoggingIntegrationTest.groovy]: :x: We'll need a test for this behavior:

> This change removes the wrapper logs for unzipping and setting
permissions, changing them to only log when something went wrong.

**Comment by eriwen** [comment]
Superseded by #4767

**Pull request closed without merging**